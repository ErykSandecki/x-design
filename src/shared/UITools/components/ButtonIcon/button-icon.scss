.ButtonIcon {
  @include theme-selector(
    'neutral-3',
    background-color,
    '&:hover:not(.ButtonIcon--selected)'
  );

  align-items: center;
  border-radius: 5px;
  cursor: pointer;
  display: flex;
  height: 24px;
  justify-content: center;
  width: 24px;

  &--selected {
    &--light {
      background-color: rgba($color: $--xd-blue-1--light, $alpha: 30%);
    }

    &--dark {
      background-color: rgba($color: $--xd-blue-1--dark, $alpha: 30%);
    }
  }
}
