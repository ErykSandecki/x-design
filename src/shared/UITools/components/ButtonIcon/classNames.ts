// utils
import { composeClassNames } from 'utils';

export const className = 'ButtonIcon';

export const classNames = composeClassNames(className, [className, 'selected']);
