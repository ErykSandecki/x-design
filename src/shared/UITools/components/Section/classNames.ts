//
import { composeClassNames } from 'utils';

export const className = 'Section';

export const classNames = composeClassNames(className, [className] as const, ['label'] as const);
