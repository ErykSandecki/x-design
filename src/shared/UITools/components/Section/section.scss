.Section {
  @include theme('neutral-3', border-bottom, '1px solid');

  padding: 12px 8px 12px 16px;

  &__header {
    align-items: center;
    display: flex;
    justify-content: space-between;
    margin-bottom: 10px;
  }

  &__label {
    @include theme('neutral-1', color);

    align-items: center;
    display: flex;
    font-size: 11px;
    user-select: none;
  }

  &__buttons {
    align-items: center;
    column-gap: 2.5px;
    display: flex;
    justify-content: center;
  }
}
