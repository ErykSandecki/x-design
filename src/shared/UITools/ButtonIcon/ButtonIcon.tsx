import { FC, useRef } from 'react';
import { noop } from 'lodash';

// components
import Box, { TBoxProps } from '../../UI/Box/Box';
import ButtonIconPopover from './ButtonIconPopover/ButtonIconPopover';
import Icon, { TIconProps } from '../../UI/Icon/Icon';
import Tooltip, { TTooltipProps } from '../../UI/Tooltip/Tooltip';

// hooks
import { useClickEvent } from './hooks/useClickEvent';
import { useOutsideClick, useTheme } from 'hooks';

// others
import { className, classNames } from './classNames';

// styles
import styles from './button-icon.scss';

// types
import { ColorsTheme, E2EAttribute } from 'types';
import { TE2EDataAttributeProps } from '../../E2EDataAttributes/E2EDataAttribute';
import { TPopoverProps } from '../Popover/Popover';

export type TSectionProps = Pick<TIconProps, 'name'> &
  TBoxProps & {
    e2eValue?: TE2EDataAttributeProps['value'];
    idContainer?: string;
    popoverChildren?: TPopoverProps['children'];
    popoverOffset?: TPopoverProps['offset'];
    selected?: boolean;
    tooltip?: Omit<TTooltipProps, 'children'>;
  };

export const ButtonIcon: FC<TSectionProps> = ({
  e2eValue = '',
  idContainer,
  name,
  onClick,
  popoverChildren,
  popoverOffset,
  selected = false,
  tooltip,
  ...restProps
}) => {
  const ref = useRef(null);
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);
  const { selected: selectedPopover, setSelected: setSelectedPopover } = useOutsideClick([ref], ref, noop, idContainer);
  const isSelected = selectedPopover || selected;
  const onClickHandler = useClickEvent(onClick, selectedPopover, setSelectedPopover);

  return (
    <Tooltip ref={ref} {...tooltip}>
      <Box
        classes={{
          className: cx(classNamesWithTheme[className].name, [
            classNamesWithTheme[className].modificators.selected,
            isSelected,
          ]),
        }}
        e2eAttribute={E2EAttribute.buttonIcon}
        e2eValue={e2eValue}
        onClick={onClickHandler}
        {...restProps}
      >
        <Icon color={selected ? ColorsTheme.blue1 : ColorsTheme.neutral1} height={14} name={name} width={14} />
        {popoverChildren && (
          <ButtonIconPopover
            ref={ref}
            selected={selectedPopover}
            setSelected={setSelectedPopover}
            offset={popoverOffset}
          >
            {popoverChildren}
          </ButtonIconPopover>
        )}
      </Box>
    </Tooltip>
  );
};

export default ButtonIcon;
