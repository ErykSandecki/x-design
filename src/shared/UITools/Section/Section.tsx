import { FC, memo, ReactNode } from 'react';

// components
import Box from '../../UI/Box/Box';
import { Small } from '../../UI/Typography';

// hooks
import { useTheme } from 'hooks';

// others
import { className, classNames } from './classNames';

// styles
import styles from './section.scss';

// types
import { E2EAttribute } from 'types';
import { TE2EDataAttributeProps } from '../../E2EDataAttributes/E2EDataAttribute';

export type TSectionProps = {
  buttonsIcon?: Array<ReactNode>;
  children: ReactNode;
  e2eValue?: TE2EDataAttributeProps['value'];
  label?: string;
};

export const Section: FC<TSectionProps> = ({ buttonsIcon = [], children, e2eValue = '', label }) => {
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);

  return (
    <Box
      classes={{ className: cx(classNamesWithTheme[className]) }}
      e2eAttribute={E2EAttribute.section}
      e2eValue={e2eValue}
    >
      {label && (
        <Box sx={{ alignItems: 'center', display: 'flex', justifyContent: 'space-between', mb: 10 }}>
          <Small classes={{ className: cx(classNamesWithTheme.label) }}>{label}</Small>
          <Box sx={{ alignItems: 'center', columnGap: '2.5px', display: 'flex', justifyContent: 'center' }}>
            {buttonsIcon.map((buttonIcon) => buttonIcon)}
          </Box>
        </Box>
      )}
      {children}
    </Box>
  );
};

export default memo(Section);
