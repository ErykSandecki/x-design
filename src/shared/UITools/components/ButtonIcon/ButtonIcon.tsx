import { FC } from 'react';

// components
import Box, { TBoxProps } from '../../../UI/components/Box/Box';
import Icon, { TIconProps } from '../../../UI/components/Icon/Icon';
import Tooltip, { TTooltipProps } from '../../../UI/components/Tooltip/Tooltip';

// hooks
import { useTheme } from 'hooks';

// others
import { className, classNames } from './classNames';

// styles
import styles from './button-icon.scss';

// types
import { ColorsTheme, E2EAttribute } from 'types';
import { TE2EDataAttributeProps } from '../../../E2EDataAttributes/E2EDataAttribute';

export type TSectionProps = Pick<TIconProps, 'name'> &
  TBoxProps & {
    e2eValue?: TE2EDataAttributeProps['value'];
    selected?: boolean;
    tooltip?: Omit<TTooltipProps, 'children'>;
  };

export const ButtonIcon: FC<TSectionProps> = ({ e2eValue = '', name, selected = false, tooltip, ...restProps }) => {
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);

  return (
    <Tooltip {...tooltip}>
      <Box
        classes={{
          className: cx(classNamesWithTheme[className].name, [
            classNamesWithTheme[className].modificators.selected,
            selected,
          ]),
        }}
        e2eAttribute={E2EAttribute.buttonIcon}
        e2eValue={e2eValue}
        {...restProps}
      >
        <Icon color={selected ? ColorsTheme.blue1 : ColorsTheme.neutral1} height={14} name={name} width={14} />
      </Box>
    </Tooltip>
  );
};

export default ButtonIcon;
