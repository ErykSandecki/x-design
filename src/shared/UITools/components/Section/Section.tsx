import { FC, ReactNode } from 'react';

// components
import Box from '../../..//UI/components/Box/Box';
import { Small } from '../../../UI/components/Typography';

// hooks
import { useTheme } from 'hooks';

// others
import { className, classNames } from './classNames';

// styles
import styles from './section.scss';

// types
import { E2EAttribute } from 'types';

export type TSectionProps = {
  buttonsIcon?: Array<ReactNode>;
  children: ReactNode;
  label?: string;
};

export const Section: FC<TSectionProps> = ({ buttonsIcon = [], children, label }) => {
  const { classNamesWithTheme, cx } = useTheme(classNames, styles);

  return (
    <Box classes={{ className: cx(classNamesWithTheme[className]) }} e2eAttribute={E2EAttribute.section}>
      {label && (
        <div className={cx(classNamesWithTheme.header)}>
          <Small classes={{ className: cx(classNamesWithTheme.label) }}>{label}</Small>
          <div className={cx(classNamesWithTheme.buttons)}>{buttonsIcon.map((buttonIcon) => buttonIcon)}</div>
        </div>
      )}
      {children}
    </Box>
  );
};

export default Section;
