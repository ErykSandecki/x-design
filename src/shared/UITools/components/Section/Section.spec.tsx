import { render } from '@testing-library/react';

// components
import ButtonIcon from '../ButtonIcon/ButtonIcon';
import Section from './Section';

describe('Section snapshots', () => {
  it('should render Section', () => {
    // before
    const { asFragment } = render(<Section label="label">children</Section>);

    // result
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render with buttons', () => {
    // before
    const { asFragment } = render(
      <Section buttonsIcon={[<ButtonIcon name="Close" key={0} />]} label="label">
        children
      </Section>,
    );

    // result
    expect(asFragment()).toMatchSnapshot();
  });
});
