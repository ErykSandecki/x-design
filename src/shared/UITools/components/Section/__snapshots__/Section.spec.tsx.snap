// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section snapshots should render Section 1`] = `
<DocumentFragment>
  <div
    class="css-wcnq0t Section Section--light"
    data-test-section=""
  >
    <div
      class="Section__header Section__header--light"
    >
      <small
        class="css-wcnq0t Section__label Section__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
        data-test-text=""
        style="text-align: inherit;"
      >
        label
      </small>
      <div
        class="Section__buttons Section__buttons--light"
      />
    </div>
    children
  </div>
</DocumentFragment>
`;

exports[`Section snapshots should render with buttons 1`] = `
<DocumentFragment>
  <div
    class="css-wcnq0t Section Section--light"
    data-test-section=""
  >
    <div
      class="Section__header Section__header--light"
    >
      <small
        class="css-wcnq0t Section__label Section__label--light Typography Typography--light Typography--neutral1 Typography--neutral1--light Typography--regular Typography--regular--light Typography--normal Typography--normal--light Typography--small Typography--small--light"
        data-test-text=""
        style="text-align: inherit;"
      >
        label
      </small>
      <div
        class="Section__buttons Section__buttons--light"
      >
        <div
          class="css-wcnq0t ButtonIcon ButtonIcon--light"
          data-test-button-icon=""
        >
          <div
            class="css-wcnq0t Icon Icon--light Icon--neutral1 Icon--neutral1--light"
            data-test-icon="close"
            height="14"
            name="Close"
            width="14"
          />
        </div>
      </div>
    </div>
    children
  </div>
</DocumentFragment>
`;
